### PR TITLE
Ran-commands for the roller

### PR DESCRIPTION
**Pull request by RJ0907**

### CODE REVIEW

File: src/main/java/edu/greenblitz/robotName/commands/arm/roller/RollerCommand.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package edu.greenblitz.robotName.commands.arm.roller;
+
+import edu.greenblitz.robotName.subsystems.arm.Roller;
+import edu.greenblitz.robotName.utils.GBCommand;
+
+public class RollerCommand extends GBCommand {
```
Comment (aaa-png) [inline src/main/java/edu/greenblitz/robotName/commands/arm/roller/RollerCommand.java:6]: abstract

File: src/main/java/edu/greenblitz/robotName/commands/arm/roller/RunRollerReverse.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+public class RunRollerReverse extends RollerCommand{
+    @Override
+    public void initialize() {
+        roller.rollOut();
+    }
+
+    @Override
+    public boolean isFinished() {
+        return !Wrist.getInstance().isObjectInside();
```
Comment (aaa-png) [inline src/main/java/edu/greenblitz/robotName/commands/arm/roller/RunRollerReverse.java:13]: are you sure this is the right condition?

File: src/main/java/edu/greenblitz/robotName/commands/arm/roller/RunRoller.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package edu.greenblitz.robotName.commands.arm.roller;
+
+import edu.greenblitz.robotName.subsystems.arm.EndEffector.RollerUtils.RollerConstants;
+import edu.greenblitz.robotName.subsystems.arm.Wrist;
+
+public class RunRoller extends RollerCommand{
```
Comment (aaa-png) [inline src/main/java/edu/greenblitz/robotName/commands/arm/roller/RunRoller.java:6]: i would add to both classes a time constant for safty. add to the is finiehd once the object is out, wait half a sec or smth to make sure its actually out

**Review by aaa-png — CHANGES_REQUESTED** [review]
Romy from amirs user

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]
listen to amir, and good job!

File: src/main/java/edu/greenblitz/robotName/commands/arm/roller/ReverseRunRoller.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+public class ReverseRunRoller extends RollerCommand{
+    @Override
+    public void initialize() {
+        roller.rollOut();
+    }
+
+    @Override
+    public boolean isFinished() {
+        return Wrist.getInstance().isObjectInside();
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/commands/arm/roller/ReverseRunRoller.java:13]: move to another command. the reverse run roller shouldn't be depending on the beam breaker

File: src/main/java/edu/greenblitz/robotName/commands/arm/roller/RunRoller.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    @Override
+    public void initialize() {
+        roller.rollIn();
+    }
+
+    @Override
+    public boolean isFinished() {
+        if (!Wrist.getInstance().isObjectInside()){
+            new WaitCommand(0.5);
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/commands/arm/roller/RunRoller.java:16]: what is 0.5? if it is a constant move to constants file. (something like .../IntakeConstants -> TIME_AFTER_OBJECT_IN_ROBOT)

File: src/main/java/edu/greenblitz/robotName/commands/arm/roller/RunRoller.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+public class RunRoller extends RollerCommand{
+    @Override
+    public void initialize() {
+        roller.rollIn();
+    }
+
+    @Override
+    public boolean isFinished() {
+        if (!Wrist.getInstance().isObjectInside()){
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/commands/arm/roller/RunRoller.java:15]: and what is the note will get to the shooter? add the other condition as an or condition

File: src/main/java/edu/greenblitz/robotName/commands/arm/roller/ReverseRunRoller.java (hunk 4 above)
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/commands/arm/roller/ReverseRunRoller.java:13]: and im not sure that if hte object is in the wrist is the correct function for checking if there is an object in the robot

File: src/main/java/edu/greenblitz/robotName/commands/arm/roller/RollerCommand.java [hunk 7]
```diff
@@ -0,0 +1,13 @@
+package edu.greenblitz.robotName.commands.arm.roller;
+
+import edu.greenblitz.robotName.subsystems.arm.Roller;
+import edu.greenblitz.robotName.utils.GBCommand;
+
+public abstract class RollerCommand extends GBCommand {
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/commands/arm/roller/RollerCommand.java:6]: suggestion:
add a command run in and run out with depending on the beam breaker. the run and reverse run commands should only run the motor and another command should decide if to stop it.

**Review by NoamRosenberg08 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/robotName/commands/arm/roller/RollerCommand.java (hunk 7 above)
Comment (tnrz135) [reply src/main/java/edu/greenblitz/robotName/commands/arm/roller/RollerCommand.java:6]: It is not only a suggestion, it is a comment now

File: src/main/java/edu/greenblitz/robotName/commands/arm/roller/ShootFromRollerToAmp.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+public class ShootFromRollerToAmp extends RollerCommand{
+    @Override
+    public void initialize() {
+        roller.rollIn();
+    }
+
+    @Override
+    public boolean isFinished() {
+        if (!Wrist.getInstance().isObjectInside()){
+            new WaitCommand(RollerConstants.TIME_AFTER_OBJECT_IN_ROBOT);
```
Comment (tnrz135) [inline src/main/java/edu/greenblitz/robotName/commands/arm/roller/ShootFromRollerToAmp.java:16]: This wouldn't work, you are creating a new instance of a command, you need to schedule it and wait for it to finish. It should be simpler to simply check for the time in the isFinished function.

**Review by tnrz135 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/robotName/commands/arm/roller/RollerCatchNote.java [hunk 9]
```diff
@@ -0,0 +1,20 @@
+package edu.greenblitz.robotName.commands.arm.roller;
+
+import edu.greenblitz.robotName.subsystems.arm.Wrist;
+
+public class RollerCatchNote extends RollerCommand{
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/commands/arm/roller/RollerCatchNote.java:5]: consider changing the name to RunRollerUntilNoteInside

File: src/main/java/edu/greenblitz/robotName/commands/arm/roller/ShootFromRollerToAmp.java [hunk 10]
```diff
@@ -0,0 +1,26 @@
+package edu.greenblitz.robotName.commands.arm.roller;
+
+import edu.greenblitz.robotName.subsystems.arm.EndEffector.RollerUtils.RollerConstants;
+import edu.greenblitz.robotName.subsystems.arm.Wrist;
+import edu.wpi.first.wpilibj2.command.WaitCommand;
+
+public class ShootFromRollerToAmp extends RollerCommand{
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/commands/arm/roller/ShootFromRollerToAmp.java:7]: consider renaming to ReleaseNote

File: src/main/java/edu/greenblitz/robotName/commands/arm/roller/ShootFromRollerToAmp.java (hunk 8 above)
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/commands/arm/roller/ShootFromRollerToAmp.java:16]: wont work. consider adding .schedule.
but still, it won't work. the new WaitCommand is creating a new thread to run while other commands run so it won't stop the current command and wait 0.5 seconds. you can either do what @tnrz135 suggests or make a sequential command group.

File: src/main/java/edu/greenblitz/robotName/subsystems/arm/EndEffector/WristUtils/WristConstants.java [hunk 11]
```diff
@@ -8,7 +8,7 @@ public class WristConstants {
     public enum PresetPositions {
 
         STARTING(Rotation2d.fromDegrees(90)),
-        SCORE(Rotation2d.fromDegrees(-45)),
+        SCORE_TO_AMP(Rotation2d.fromDegrees(-45)),
         TRANSFER(Rotation2d.fromDegrees(180));
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/arm/EndEffector/WristUtils/WristConstants.java:12]: consider tenaming to TRANSFER_POSITION, SCORE_TO_AMP_POSITION STARTING_POSITION

**Review by NoamRosenberg08 — CHANGES_REQUESTED** [review]

**Comment by bossyshrimp68** [comment]
lil bit of walla balla

File: src/main/java/edu/greenblitz/robotName/commands/arm/roller/RunRoller.java (hunk 6 above)
Comment (greenblitz4590) [reply src/main/java/edu/greenblitz/robotName/commands/arm/roller/RunRoller.java]: image

File: src/main/java/edu/greenblitz/robotName/subsystems/arm/EndEffector/RollerUtils/RollerConstants.java [hunk 12]
```diff
@@ -4,6 +4,7 @@ public class RollerConstants {
 
 	public static final double ROLL_FORWARD_POWER = 1;
 	public static final double ROLL_BACKWARD_POWER = -1;
+	public static final double TIME_AFTER_OBJECT_IN_ROBOT = 0.5;
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/arm/EndEffector/RollerUtils/RollerConstants.java:7]:
```suggestion
	public static final double TIME_UNTIL_NOTE_EXIT_ARM= 0.5;
```
not sure if i understood, but if i did change the name to this if not to a different one

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]
AMAZING JOB!!
you really improved, just make sure you check in the simulation 
(use shuffleboard for the isIn and stuff)

File: src/main/java/edu/greenblitz/robotName/subsystems/arm/EndEffector/RollerUtils/RollerConstants.java [hunk 13]
```diff
@@ -4,6 +4,7 @@ public class RollerConstants {
 
 	public static final double ROLL_FORWARD_POWER = 1;
 	public static final double ROLL_BACKWARD_POWER = -1;
+	public static final double TIME_UNTIL_NOTE_EXIT_ARM = 0.5;
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/arm/EndEffector/RollerUtils/RollerConstants.java:7]:
```suggestion
	public static final double TIME_UNTIL_NOTE_EXIT_ROLLER = 0.5;
```
sorry to do this againnnn

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]

**Review by bossyshrimp68 — APPROVED** [review]

File: src/main/java/edu/greenblitz/robotName/commands/arm/roller/ReleaseNote.java [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+public class ReleaseNote extends RollerCommand{
+    @Override
+    public void initialize() {
+        roller.rollIn();
+    }
+
+    @Override
+    public boolean isFinished() {
+        if (!Wrist.getInstance().isObjectInside()){
+            new WaitCommand(RollerConstants.TIME_UNTIL_NOTE_EXIT_ROLLER).schedule();
```
Comment (tnrz135) [inline src/main/java/edu/greenblitz/robotName/commands/arm/roller/ReleaseNote.java:16]: This won't wait for the WaitCommand, use Sequential Command

**Review by tnrz135 — CHANGES_REQUESTED** [review]

**Review by tnrz135 — APPROVED** [review]

File: src/main/java/edu/greenblitz/robotName/commands/arm/roller/ReleaseNote.java [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+import edu.greenblitz.robotName.subsystems.arm.roller.RollerConstants;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import edu.wpi.first.wpilibj2.command.WaitCommand;
+
+public class ReleaseNote extends SequentialCommandGroup {
+
+    public ReleaseNote(){
+        super(
+                new RunRollerUntilObjectOut(),
+                new WaitCommand(RollerConstants.TIME_UNTIL_NOTE_EXIT_ROLLER),
```
Comment (roygold8) [inline src/main/java/edu/greenblitz/robotName/commands/arm/roller/ReleaseNote.java:12]: Why do we need to wait? Roll until note is out should do that

File: src/main/java/edu/greenblitz/robotName/commands/arm/roller/RunRoller.java [hunk 16]
```diff
@@ -0,0 +1,13 @@
+package edu.greenblitz.robotName.commands.arm.roller;
+
+public class RunRoller extends RollerCommand{
```
Comment (roygold8) [inline src/main/java/edu/greenblitz/robotName/commands/arm/roller/RunRoller.java:3]: Name is not indicative

**Review by roygold8 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/robotName/subsystems/arm/roller/RollerConstants.java [hunk 17]
```diff
@@ -4,6 +4,7 @@ public class RollerConstants {
 
 	public static final double ROLL_FORWARD_POWER = 1;
 	public static final double ROLL_BACKWARD_POWER = -1;
+	public static final double SAFETY_TIME_DISTANCE_AFTER_NOTE_RELEASED = 0.5;
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/subsystems/arm/roller/RollerConstants.java:7]: time distance? either time or distance

File: src/main/java/edu/greenblitz/robotName/commands/arm/roller/RunRollerUntilNoteInside.java [hunk 18]
```diff
@@ -0,0 +1,18 @@
+package edu.greenblitz.robotName.commands.arm.roller;
+
+public class RunRollerUntilNoteInside extends RollerCommand{
+    @Override
+    public void initialize() {
+        roller.rollOut();
```
[the remark this answers is not in the document]
Comment (RJ0907) [reply src/main/java/edu/greenblitz/robotName/commands/arm/roller/RunRollerUntilNoteInside.java:6]: roll out is the right function to be called

File: src/main/java/edu/greenblitz/robotName/commands/arm/roller/RunRollerUntilObjectOut.java [hunk 19]
```diff
@@ -0,0 +1,13 @@
+package edu.greenblitz.robotName.commands.arm.roller;
+
+public class RunRollerUntilObjectOut extends RollerCommand{
+    @Override
+    public void initialize() {
+        roller.rollIn();
```
[the remark this answers is not in the document]
Comment (RJ0907) [reply src/main/java/edu/greenblitz/robotName/commands/arm/roller/RunRollerUntilObjectOut.java:6]: roll in is the right function to be called

File: src/main/java/edu/greenblitz/robotName/commands/arm/roller/ReleaseNote.java (hunk 15 above)
Comment (aaa-png) [reply src/main/java/edu/greenblitz/robotName/commands/arm/roller/ReleaseNote.java]: to make it sure

**Review by yonnbro — APPROVED** [review]

File: src/main/java/edu/greenblitz/robotName/commands/arm/roller/ReleaseNote.java [hunk 20]
```diff
@@ -0,0 +1,16 @@
...
+
+import edu.greenblitz.robotName.subsystems.arm.roller.RollerConstants;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import edu.wpi.first.wpilibj2.command.WaitCommand;
+
+public class ReleaseNote extends SequentialCommandGroup {
+
+    public ReleaseNote(){
+        super(
+                new RunRollerUntilObjectOut(),
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/commands/arm/roller/ReleaseNote.java:11]: shouldn't it stop the roller?

**Review by aaa-png — APPROVED** [review]
hillel - romy approved so I am approving just for it not to be changes requested

**Pull request closed without merging**